### DISCUSSION TITLE
Overridden date dependency not having an effect on previews

### DISCUSSION THREAD

**Discussion by LudvigIversen** [pointfreeco/swift-dependencies#121] [Q&A]
### Description

Let's say as an example that we have this view and model;

```swift
import Dependencies
import SwiftUI

struct ExampleView: View {
    let model = Model()

    var body: some View {
        VStack {
            Text(model.date.description)
                .foregroundColor(model.isBeforeToday ? .blue : .red )
        }
    }
}

struct Model {
    var date = Date("2023-09-25")!
    var isBeforeToday: Bool {
        @Dependency(\.date.now) var now
        return date.isBeforeDate(now, granularity: .day)
    }
}
```
Here, the text in the view should be blue if the models date is before todays date. In this case the text would be blue since the day of writing is 2023-09-26. However I also want to see that the logic for the text is correct and that it can turn red. Normally this date would not be hardcoded, but gotten from an API or similar. So, I create a preview, like so;

``` swift
struct ExampleView_Previews: PreviewProvider {
    static var previews: some View {
        withDependencies {
            $0.date.now = Date("2023-01-23")!
        } operation: {
            ExampleView()
        }
    }
}
```

However, the date would still be blue, when I believe it should be red. The dependency determining what todays date is does not get overwritten. 

### Checklist

- [x] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-dependencies/issues) or [discussion](https://github.com/pointfreeco/swift-dependencies/discussions).

### Expected behavior

The date dependency should be overwritten by the preview and, in this case, the text should be displayed as red.

### Actual behavior

The dependency is not overwritten, and the text remains blue. Instead of using the value I provide in the preview, it instead uses the standard value, that being;
```swift
static let testValue = DateGenerator {
      XCTFail(#"Unimplemented: @Dependency(\.date)"#)
      return Date()
 }
```

An important note to add, overriding the dependency works in unit tests, the logic also works when compiling normally/in production. 

### Steps to reproduce

Reuse the example code given here above. 

### Xcode version information

15.0

**Comment by stephencelis** [comment]
@LudvigIversen In your example, you are overriding the dependencies of the initialization of `ExampleView`, but no dependencies are accessed during initialization. Instead, `date` is accessed when the body of the view renders (which happens outside of the `withDependencies` block).

One thing you can do is hold onto the `@Dependency` at the model level, so that the dependencies supplied are held onto for the lifetime of the model:

```diff
 struct Model {
+    @Dependency(\.date.now) var now
     var date = Date("2023-09-25")!
     var isBeforeToday: Bool {
-        @Dependency(\.date.now) var now
         return date.isBeforeDate(now, granularity: .day)
     }
 }
```

This should fix things.

Since this isn't a bug with the library, and more a question of use, I'm going to convert to a discussion.
